src/qiboml/models/utils.py

```diff
@@ -142,6 +151,58 @@ def data_regression(
         log.info(f"Obtained noise map params: {self._mitigation_map_popt}.")
 
 
+@dataclass
+class Calibrator:
+
+    runcard: Runcard
+    """Qibocal runcard with the calibration protocol to be executed."""
+    path: Path
+    """Folder to dump the protocol's data and results."""
+    backend: Backend
+    """Qibo backend."""
+    trigger_shots: int = 100
+    """NUmber of shots to trigger :meth:`qiboml.models.utils.Calibrator.execute_experiments`"""
+    _history: list[History] = field(default_factory=list)
+    _counter: int = 0
+
+    @property
+    def history(self):
+        return self._history
+
+    def __call__(self):
+        self._counter += 1
+        if self._counter % self.trigger_shots == 0:
+            self.execute_experiments()
+
+    def execute_experiments(self):
+        """Execute the experiments in the runcard."""
+        platform = self.backend.platform
+        assert platform is not None, "Invalid None platform"
```

Comment:
```suggestion
        assert platform is not None
```
I think that the error message is redundant in this case.